lesson-5/chapter-6/zombieownership.sol

```diff
@@ -23,7 +23,7 @@ contract ZombieOwnership is ZombieAttack, ERC721 {
   }
 
   function transferFrom(address _from, address _to, uint256 _tokenId) external payable {
-    require (zombieToOwner[_tokenId] == msg.sender || zombieApprovals[_tokenId] == msg.sender);
+    require ((zombieToOwner[_tokenId] == msg.sender && _from == msg.sender) || (zombieApprovals[_tokenId] == msg.sender && _to == msg.sender));
```

Comment:
I think in both cases you need to check `zombieToOwner[_tokenId] == _from`.

So the fix would be:

`require (zombieToOwner[_tokenId] == _from && (msg.sender == _from || (msg.sender == _to && zombieApprovals[_tokenId] == msg.sender));`